### DISCUSSION TITLE
How i can get featureset name from step/scenario?

### DISCUSSION THREAD

**Discussion by TkachevAlex** [behave/behave#1098] [Q&A] [answered]
Hi! I need know featureset name from step or scenario. How i can know it?

**Comment by jenisys** [comment]
I am not quite sure that I understand what you are looking for.
Do you mean:

* the Feature name (or title) ?!?
* The filename of the feature file that contains a scenario or step ?!?
* … (something else)

And explain why you need that information

**Reply by TkachevAlex** [reply]
My starting line is "behave smoke.featureset". In featureset file are some feature files with smoke tests. And i would like print featureset name in my test log file

**Reply by jenisys** [reply, marked as answer]
Ok, the correct command line is: `behave @smoke.featureset` (by using AT-files that contains the features that you like to run).

Currently, you can not determine „smoke“ from the featureset-file.
My suggestion is to use either 

* an environment variable or 
* a Userdata parameter 

to pass this information (if you really need it during the test run).

NOTE:
But you need to decide what you do, if two AT-files are used (for example) or additional feature-files are provided on the command-line next to the AT-file.
